hmt_escrow/storage.py

```diff
@@ -107,16 +114,108 @@ def upload(msg: Dict, public_key: bytes) -> Tuple[str, str]:
 
     hash_ = hashlib.sha1(manifest_.encode('utf-8')).hexdigest()
     try:
-        key = IPFS_CLIENT.add_bytes(_encrypt(public_key, manifest_))
+        ipfsFileHash = IPFS_CLIENT.add_bytes(_encrypt(public_key, manifest_))
     except Exception as e:
         LOG.warning("Adding bytes with IPFS failed because of: {}".format(e))
         raise e
-    return hash_, key
+
+    if ipns_keypair_name != '':
+        try:
+            # publish ipns ... docs: https://ipfs.io/ipns/12D3KooWEqnTdgqHnkkwarSrJjeMP2ZJiADWLYADaNvUb6SQNyPF/docs/http_client_ref.html#ipfshttpclient.Client.name
+            # TODO: is it faster if
```

Comment:
?

Reply:
this is an api thing.. so right now `upload` does this:
 `upload(msg: Dict, public_key: bytes, ipns_keypair_name: str = '')`
 1) put file on IPFS
 2) if ipns_keypair_name, then pin to IPNS

here are two different examples to clairfy
* upload to IPFS: `upload(data, public_key, '')`
* upload to IPFS and pin to IPNS `upload(data, public_key, 'escrow1')`

altenatively we can chagne the api to be like this
`ipfs_hash = upload(data, public_key)`
`pin_ipns(ipfs_hash, 'escrow1')`

Reply:
I'm mostly just wondering what your TODO item means?

Reply:
Yea, should remove the TODO comment or finish the thought

Reply:
ah, removed.